tests/TestFormula.elm

```diff
@@ -202,6 +203,91 @@ parseSignedTests =
         , test "F a" <| \() -> testParseSigned "F a" <| F a
         ]
 
-
+--testovanie spravneho dosadenia
+testSubstitution : Substitution -> Formula -> Formula -> Test
+testSubstitution s original new =
+    test
+        ("in formula " ++ (original |> Formula.strFormula) ++" using subs " ++ (Formula.strSubstitution s))
+        (\() -> Expect.equal (Ok new) (Formula.removeQuantifierAndSubstitute s original))
+
+--testovanie spravneho vyberu premennych -> aplikovateľnosť substitúcie
+testSubstitutionFail : Substitution -> Formula -> Test
+testSubstitutionFail s original =
+    test
+    ("substitution is not applicable in " ++ Formula.strFormula original ++ " by substituting " ++ (Formula.strSubstitution s))
+        (\() -> Expect.err (Formula.removeQuantifierAndSubstitute s original) )
+
+--testovanie substitucie viacerych premennych naraz
+
+substitutionTests =
+    describe "Substitution "
+        [ testSubstitutionFail
+            (Dict.fromList [("x", Var "k")])
+            (Formula.parse "\\forall x \\forall k P(x,k)" |> Result.withDefault FF)
+        , testSubstitution
+            (Dict.fromList [("x", Var "y"), ("y", Var "x")])
+            (Atom "P" [Var "x", Var "y"]) -- P(x,y)
```

Comment:
Btw, hore su definovane vselijake helpre pre `p`, `x`, `y` a pod... (akurat pozor, ze niektore z nich robia viacpismenkove mena, napr "yyy").
Bolo by to s nimi trochu kratsie/citalnejsie (mozno)
```elm
p [x, y]
p [ f [ x ] ]
```
a pod.